### DISCUSSION TITLE
Add ability to notify users if there is an issue collecting data (eg. auth token expires and can't be revalidated)

### DISCUSSION THREAD

**Discussion by drees** [teslamate-org/teslamate#1960] [Ideas]
Given light of #1945 which cause TeslaMate to stop collecting data without warning, it would be useful to know when there is an issue with TeslaMate that causes data collection to stop.

Additionally, if we have any idea of when an auth token expires, it would be useful to get a notification ahead of time so a new token can be set up proactively.

Possible notification methods:
Email, Web Notification, SMS

**Comment by nicholashead** [comment]
I think this is a valid concern. I check Teslamate periodically to make sure it's still "connected". The most convenient thing for me would be email or Pushover integration for alerts.

**Comment by adriankumpf** [comment]
> Given light of #1945 which cause TeslaMate to stop collecting data without warning, it would be useful to know when there is an issue with TeslaMate that causes data collection to stop.
> 
> Additionally, if we have any idea of when an auth token expires, it would be useful to get a notification ahead of time so a new token can be set up proactively.

Note, that this only affects new users that are trying to sign in. TeslaMate won't suddenly stop collecting data.

Existing access tokens also won't expire, see https://github.com/adriankumpf/teslamate/issues/1945#issuecomment-928021708

When there are any issues, e.g. if the Tesla servers cannot be reached, the `teslamate/cars/$car_id/healthy` [MQTT topic](https://docs.teslamate.org/docs/integrations/mqtt) will report `false`. If you use HomeAssistant, for example, you can use it to get notified.
